app/Controller/NoticelistsController.php

```diff
@@ -102,29 +102,34 @@ public function update()
         }
     }
 
-    public function toggleEnable()
+    public function toggleEnable($noticelist_id = false)
     {
-        $id = $this->request->data['Noticelist']['data'];
-        if (!is_numeric($id)) {
-            return new CakeResponse(array('body' => json_encode(array('saved' => false, 'errors' => 'Noticelist not found.')), 'status' => 200, 'type' => 'json'));
-        }
-        $currentState = $this->Noticelist->find('first', array('conditions' => array('id' => $id), 'recursive' => -1));
-        if (empty($currentState)) {
-            return new CakeResponse(array('body' => json_encode(array('saved' => false, 'errors' => 'Noticelist not found.')), 'status' => 200, 'type' => 'json'));
-        }
-        $currentState['Noticelist']['ref'] = json_decode($currentState['Noticelist']['ref']);
-        $currentState['Noticelist']['geographical_area'] = json_decode($currentState['Noticelist']['geographical_area']);
-        if ($currentState['Noticelist']['enabled']) {
-            $currentState['Noticelist']['enabled'] = 0;
-            $message = 'disabled';
-        } else {
-            $currentState['Noticelist']['enabled'] = 1;
-            $message = 'enabled';
-        }
-        if ($this->Noticelist->save($currentState)) {
-            return new CakeResponse(array('body' => json_encode(array('saved' => true, 'success' => 'Noticelist ' . $message)), 'status' => 200, 'type' => 'json'));
+        if ($this->request->is('post')) {
+            $this->Noticelist->read(null, $noticelist_id);
```

Comment:
This is generally not a good idea - find queries are preferred. The inheritence model of cake will actually fetch all notice list entries in this case